File: Assignment 5/C/01. implement singly circular ll.c
#include <stdio.h>
#include <stdlib.h>

typedef struct node{
    int val;
    struct node* next;
}node;

int main(){
    node* head = (node*)malloc(sizeof(node));
    node* snd = (node*)malloc(sizeof(node));
    node* trd = (node*)malloc(sizeof(node));
    node* four = (node*)malloc(sizeof(node));

    head->val = 1;
    snd->val = 2;
    trd->val = 3;
    four->val = 4;

    head->next = snd;
    snd->next = trd;
    trd->next = four;
    four->next = head;

    node* ptr = head;
    do{
        printf("--> %d ",ptr->val);
        ptr = ptr->next;
    }while(ptr!=head);
    printf("--\n");

    return 0;
}

File: Assignment 5/Python/01. implement singly circular ll .py

class node:
    def __init__(self,val):
        self.val = val
        self.next = None

head = node(1)
snd = node(2)
trd = node(3)
four = node(4)

head.next = snd
snd.next = trd
trd.next = four
four.next = head

ptr = head
while True:
    print(ptr.val,end=" ")
    ptr = ptr.next
    if(ptr==head):
        break


# OUTPUT:
# 1 2 3 4
